tests/src/test/java/org/rhd/katapult/test/services/GithubResourceTest.java

```diff
@@ -0,0 +1,94 @@
+package org.rhd.katapult.test.services;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+
+import com.gargoylesoftware.htmlunit.ElementNotFoundException;
+import com.gargoylesoftware.htmlunit.WebClient;
+import com.gargoylesoftware.htmlunit.WebResponse;
+import com.gargoylesoftware.htmlunit.html.HtmlButton;
+import com.gargoylesoftware.htmlunit.html.HtmlInput;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import com.gargoylesoftware.htmlunit.util.NameValuePair;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.importer.ZipImporter;
+import org.jboss.shrinkwrap.api.spec.EnterpriseArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Validation of the GithubResource api endpoint
+ */
+@RunWith(Arquillian.class)
+public class GithubResourceTest {
+
+    /**
+     * Deploy the katapult.ear as built since we only test via the rest endpoints
+     * @return
+     */
+    @Deployment(testable = false)
+    public static EnterpriseArchive createDeployment() {
+        EnterpriseArchive ear = ShrinkWrap.create(EnterpriseArchive.class, "application-ear.ear")
+                .as(ZipImporter.class)
+                .importFrom(new File("../ear/target/katapult.ear"))
+                .as(EnterpriseArchive.class);
+        return ear;
+    }
+
+    @ArquillianResource
+    private URL deploymentUrl;
+
+    /**
+     * Validate that we can fork the jboss-developer/jboss-eap-quickstarts repo
+     * @throws IOException
+     */
+    @Test
+    public void should_fork_jboss_eap_quickstarts() throws IOException {
+        String forkURL = deploymentUrl.toExternalForm() + "api/github/fork?repo=jboss-developer/jboss-eap-quickstarts";
+        final WebClient webClient = new WebClient();
+
+        System.out.printf("Calling for: %s\n", forkURL);
+        HtmlPage page = webClient.getPage(forkURL);
+        WebResponse response = page.getWebResponse();
+        int statusCode = response.getStatusCode();
+        String html = page.asXml();
+        System.out.printf("Page#1 html: %s\n", html);
+        // First need to login user into github
+        String username = System.getenv("GITHUB_USERNAME");
+        String password = System.getenv("GITHUB_PASSWORD");
+        HtmlInput loginField = page.getElementById("login_field", false);
+        HtmlInput passwordField = page.getElementById("password", false);
+        HtmlInput commitBtn = page.getElementByName("commit");
+        loginField.setValueAttribute(username);
+        passwordField.setValueAttribute(password);
+        HtmlPage page2 = commitBtn.click();
+        List<NameValuePair> postParams = page2.getWebResponse().getWebRequest().getRequestParameters();
+        html = page2.asXml();
```

Comment:
Yes, they say you need to deal with 2 factor auth in the web flow page, so I will have to test it out to see what changes it would require.